### PR TITLE
feat: move trace not found to 404

### PR DESCRIPTION
**Pull request by maxdeichmann**
> [!IMPORTANT]
> Refactor `getTraceById` to return `undefined` for missing traces and update API routes to handle this with a 404 error.
> 
>   - **Behavior**:
>     - `getTraceById` in `traces.ts` returns `undefined` for missing traces instead of throwing an error.
>     - API routes in `traces.ts` throw `TRPCError` with code `NOT_FOUND` if `getTraceById` returns `undefined`.
>   - **Tests**:
>     - Updated `trace-repository.servertest.ts` to expect `undefined` instead of a thrown error when a trace is not found.
>   - **Misc**:
>     - Removed `getTraceByIdOrThrow` function from `traces.ts`.
>     - Minor logging changes in `[trpc].ts` to return the error object.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=langfuse%2Flangfuse&utm_source=github&utm_medium=referral)<sup> for 6ca1ffee72277082d8e0da9cdef8fe649dc10352. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by maxdeichmann** [comment]
image
new 404 error

File: web/src/server/api/trpc.ts [hunk 1]
```diff
@@ -121,13 +121,17 @@ const withErrorHandling = t.middleware(async ({ ctx, next }) => {
   const res = await next({ ctx }); // pass the context to the next middleware
 
   if (!res.ok) {
```
[the remark this answers is not in the document]
Comment (maxdeichmann) [reply web/src/server/api/trpc.ts:123]: If ok is false, the error exists.

**Comment by maxdeichmann** [comment]
image

Internal error still looks like this

**Review by Steffen911 — APPROVED** [review]

File: web/src/pages/api/trpc/[trpc].ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ export default createNextApiHandler({
       error,
     );
     traceException(error);
+    return error;
```
[the remark this answers is not in the document]
Comment (maxdeichmann) [reply web/src/pages/api/trpc/[trpc].ts:21]: Agree.

**Pull request merged**